### PR TITLE
Add torch.utils to the docs page, remove dead code and fix docstrings

### PR DESCRIPTION
**Pull request by albanD**
As per title.
Note that the c++ side code for the minidumps part was removed. So trying to call any of these 3 functions today results in an error saying that `torch._C` doesn't have these attributes.

### CODE REVIEW

File: torch/utils/backend_registration.py [hunk 1]
```diff
@@ -27,61 +27,66 @@ def rename_privateuse1_backend(backend_name: str) -> None:
 
     Note: this API can only be called once per process. Attempting to change
     the external backend after it's already been set will result in an error.
+    ddd
```
Comment (janeyx99) [inline torch/utils/backend_registration.py:30]: What does this do?

File: torch/utils/backend_registration.py [hunk 2]
```diff
@@ -279,6 +277,13 @@ def generate_methods_for_privateuse1_backend(for_tensor: bool = True, for_module
     and so will not be properly typed. For Storage methods generate, if you need to support sparse data storage,
     you need to extend the implementation yourself.
 
+    Args:
+        for_tensor (bool): whether register related methods for torch.Tensor class.
+        for_module (bool): whether register related methods for torch.nn.Module class.
+        for_storage (bool): whether register related methods for torch.Storage class.
+        unsupported_dtype(List[torch.dtype]): takes effect only when the storage method needs to be generated,
```
Comment (janeyx99) [inline torch/utils/backend_registration.py:284]:
```suggestion
        unsupported_dtype (List[torch.dtype]): takes effect only when the storage method needs to be generated,
```

nit: consistency

**Review by janeyx99 — APPROVED** [review]
Looks alright, curious about the ddds

File: torch/utils/backend_registration.py (hunk 1 above)
Comment (albanD) [reply torch/utils/backend_registration.py:30]: Ho that was me trying to figure out which line was causing docs build issues with very advanced techniques. Removing.

**Comment by albanD** [comment]
@huydhn this is passing CI properly here. Is that safe to reland?

**Comment by huydhn** [comment]
I will do a rebase merge.  If the job passes, it should be safe I think

**Comment by huydhn** [comment]
A rebase brings up the failure I think https://hud.pytorch.org/pr/pytorch/pytorch/105142#15300919192

**Comment by albanD** [comment]
@pytorchbot merge

All docs builds are now green!

**Pull request closed without merging**